core/models.py

```diff
@@ -132,6 +132,16 @@ def get_images(self: "Post") -> models.QuerySet:
     def update_display_counter(self: "Post") -> None:
         Post.objects.filter(pk=self.pk).update(display_counter=F("display_counter") + 1)
 
+    def children_count(self: "Post") -> int:
+        def count_descendants(post: "Post") -> int:
+            children = post.children.all()
+            total_children = children.count()
+            for child in children:
+                total_children += count_descendants(child)
+            return total_children
+
+        return count_descendants(self)
+

```

Comment:
Hmm this has potential to kill the DB.
I wonder if we should not store children amount as int on parent itself, this way we don't run count every time...
That's for future refactor. 
Nice use of recurency btw. 
```suggestion
    @property
    def children_count(self: "Post") -> int:
        def count_descendants(post: "Post") -> int:
            children = post.children.all()
            total_children = children.count()
            for child in children:
                total_children += count_descendants(child)
            return total_children

        return count_descendants(self)


```

Reply:
I'd only add this @property